### PR TITLE
fix(spawner): Normalize Task names to valid RFC 1123 names

### PR DESCRIPTION
**Pull request by BudgetSanta**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

A `TaskSpawner` with a Jira source fails to create any Task. The spawner builds each Task's name as `<spawner>-<item.ID>`, and for Jira sources `item.ID` is the issue key (e.g. `PROJECT-1234`). Kubernetes object names must be valid RFC 1123 DNS subdomains (lowercase only), so the uppercase key makes every generated name invalid and the API server rejects the create — Jira `TaskSpawner`s never produce Tasks.

GitHub issue/PR and Cron sources are unaffected because their IDs are already name-safe (numeric / timestamp).

This adds a `buildTaskName` helper and routes both task-name construction sites through it (the dedup lookup and the creation call), so the two derivations stay identical. The helper returns already-valid names unchanged and otherwise normalizes the combined name to a valid RFC 1123 name (lowercase, non-alphanumeric runs collapsed to `-`, leading/trailing separators trimmed). The raw issue key is preserved everywhere else it is used (prompt variables, browse URL, logs).

#### Which issue(s) this PR is related to:

N/A

#### Special notes for your reviewer:

- No API/CRD or generated-code changes — a spawner-side fix only.
- `buildTaskName` is a no-op for already-valid names (GitHub numeric IDs, Cron timestamps), enforced via `validation.IsDNS1123Subdomain`, so existing Task names and the name-based dedup that keys on them are unaffected. Existing spawner tests asserting names like `spawner-42` pass unchanged.
- Tests: `TestBuildTaskName` covers the Jira fix, the GitHub/Cron no-ops, and normalization edge cases; `TestRunCycleWithSource_NormalizesJiraKeyAndDedupes` runs two spawn cycles to verify normalization and that no duplicate Task is created.

#### Does this PR introduce a user-facing change?

```release-note
Fixed Jira `TaskSpawner` failing to create Tasks because issue keys produced invalid (uppercase) Kubernetes object names. Task names are now normalized to valid RFC 1123 names.
```

---
## Summary by cubic
Fixes Jira Task creation by normalizing Task names to valid RFC 1123 labels and capping them to 63 characters with a short hash when needed. GitHub and Cron names stay unchanged when already valid and within length; dedup and creation now use the same builder.

- **Bug Fixes**
  - Added `buildTaskName` to lowercase and normalize `<spawner>-<item.ID>`, collapse separators, trim, and enforce a 63-char max; on truncation, append a 10-char hash to avoid collisions.
  - Routed both dedup lookup and Task creation through `buildTaskName` to keep names consistent.
  - Added tests for Jira keys, no-op cases (GitHub/Cron), edge cases, over-length handling, and collision avoidance.

<sup>Written for commit 821970010beed8523501d5e51b01b1081acdd276. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/kelos-dev/kelos/pull/1368?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by gjkim42** [comment]
seems like a valid issue.
can you sign the CLA?

**Comment by BudgetSanta** [comment]
> seems like a valid issue. can you sign the CLA?

Yep sorry. Waiting for sign off from my company's legal team. Just didn't want to cause you any issues if i was in the wrong based on whatever my company says. If you want to duplicate the branch and move it through before I can get approval please go ahead.

**Comment by gjkim42** [comment]
@BudgetSanta 
can you take a look at cubic's review?

File: cmd/kelos-spawner/main.go [hunk 1]
```diff
@@ -807,3 +808,21 @@ func parsePollInterval(s string) time.Duration {
...
+// uppercase letters would otherwise be rejected by Kubernetes. IDs that are
+// already name-safe are returned unchanged.
+func buildTaskName(spawnerName, itemID string) string {
+	combined := spawnerName + "-" + itemID
+
+	if len(validation.IsDNS1123Subdomain(combined)) == 0 {
+		return combined
+	}
+
+	segments := strings.FieldsFunc(strings.ToLower(combined), func(r rune) bool {
```
[the remark this answers is not in the document]
Comment (BudgetSanta) [reply cmd/kelos-spawner/main.go:834]: I believe this is an unfortunate side effect of what we're trying to do here. To avoid throwing an error we're modifying the name to fix a valid name. This correction will have collision but for the most part naming like `PROJ_1` and `PROJ-1` both mapping to `-proj-1` is the desired outcome.

File: cmd/kelos-spawner/main.go [hunk 2]
```diff
@@ -807,3 +808,21 @@ func parsePollInterval(s string) time.Duration {
...
+		return !('a' <= r && r <= 'z' || '0' <= r && r <= '9')
+	})
+	return strings.Join(segments, "-")
+}
```
[the remark this answers is not in the document]
Comment (BudgetSanta) [reply cmd/kelos-spawner/main.go]: Good call out. I'll find the valid length and cut it off

**Comment by gjkim42** [comment]
@BudgetSanta 
squash commits please

**Comment by BudgetSanta** [comment]
> @BudgetSanta squash commits please

Can we not just squash on merge?

**Comment by gjkim42** [comment]
But I expect authors to organize their PR into meaningful commits and have a useful commit message.

Squash on merge may ends up having noisy commit messages by default.
